### ISSUE TITLE
[FEATURE] Add `no-empty-headings` rule

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#355]
**Is your feature request related to a problem? Please describe.**
Headings relay the structure of a webpage and provide a meaningful, hierarchical order of its content. If headings are empty or its text contents are inaccessible, this could confuse users or prevent them accessing sections of interest (https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-empty-headings.md)

**incorrect**
```html
<h1></h1>
<div role="heading" aria-level="1"></div>
<h2><span aria-hidden="true">Inaccessible text</span></h2>
```

**correct**
```html
<h1>Heading Content</h1>
<h2><span>Text</span><h2>
<div role="heading" aria-level="1">Heading Content</div>
<h3 aria-hidden="true">Heading Content</h3>
<h4 hidden>Heading Content</h4>
```

**Additional context**
 - https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-empty-headings.md

**Comment by Amarjeet325** [comment]
Hi,
I am proficient in   Full_Stack  development.
I have contributed many starknet ,stellar projects .. .
please /assign Me this issue .
Thank You !!

**Comment by ivioje** [comment]
Hello!

I’d love to help fix the issue about empty or inaccessible headings. Making sure headings are meaningful and accessible is really important for usability, and I think this rule has a lot of value.

I’m planning to:

1. Flag <h*> tags that don’t have accessible text.

2. Catch cases like role="heading" with aria-level where the content is missing or hidden.

3. Add helpful examples in the tests and documentation, both what to avoid and what’s considered valid.

**Comment by yeonjuan** [comment]
Hi @ivioje Thanks! If you have any questions along the way, feel free to reach out anytime.

**Comment by ivioje** [comment]
> Hi [@ivioje](https://github.com/ivioje) Thanks! If you have any questions along the way, feel free to reach out anytime.

Sure! Thank you.
